web-share/share-null-manual.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share null and undefined values</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="resources/manual-helper.js"></script>
+  </head>
+  <body>
+    <script>
+        setup({explicit_timeout: true});
+
+        // Expect null to be converted into the string 'null'. (This isn't good
+        // practice, but it's how Web IDL specifies conversion.) On the other
```

Comment:
> well, `<a href=""></a>`, no?

Yes, this came about sort of organically from implementing it in Chrome which took care of this for us. Basically anywhere on the web where a URL is expected, it can be relative and will be resolved to the document URL, which if it's "" means the current page URL. See `fetch`, for example.

> I tend to disagree with @foolip, here

That was my initial reaction too but this isn't really a matter of opinion for an individual standard. It's just how IDL works. IDL distinguishes optional from nullable. You don't use nullable unless you have a separate meaning for null that isn't simply "not given". And if an IDL interface doesn't specify nullable, then the `null` value is treated like any other object and converted to the required type.

As an example, [`fetch`](https://fetch.spec.whatwg.org/#fetch-method) is defined with a non-nullable argument, so if you pass null to it, it's equivalent to the string `'null'`:

    > fetch(null).then(response => console.log(response.url))
    https://github.com/w3c/web-platform-tests/pull/null